testerum-backend/testerum-runner/testerum-runner-junit/src/main/kotlin/com/testerum/runner/junit5/provider/JunitTestProvider.kt

```
@@ -0,0 +1,198 @@
+
+class JunitTestProvider(repositoryDirectory: Path,
+                        variablesEnvironment: String? = null,
+                        variableOverrides: Map<String, String> = emptyMap(),
+                        settingsFile: Path? = null,
+                        settingOverrides: Map<String, String> = emptyMap(),
+                        testPaths: List<TestPath> = emptyList(),
+                        tagsToInclude: List<String> = emptyList(),
+                        tagsToExclude: List<String> = emptyList()
+) {
+
+    private val bootstrapper: RunnerModuleBootstrapper
+    private val cmdlineParams: CmdlineParams = CmdlineParams(
+            verbose = false,
+            repositoryDirectory = repositoryDirectory,
+            variablesEnvironment = variablesEnvironment,
+            variableOverrides = variableOverrides,
+            settingsFile = settingsFile,
+            settingOverrides = settingOverrides,
+            testPaths = testPaths,
+            tagsToInclude = tagsToInclude,
+            tagsToExclude = tagsToExclude,
+            reportsWithProperties = listOf("JUNIT"),
+            managedReportsDir = null,
+            executionName = null
+    )
+
+    private lateinit var testSuiteToBeExecuted: RunnerSuite
+    private val loggingListener = JUnitEventLogger()
+
+    private lateinit var eventQueue: LinkedBlockingDeque<RunnerEvent>
+    private val eventQueueInitialized = CountDownLatch(1)
+
+    private var isTestStarted = false
+
+    init {
+        bootstrapper = RunnerModuleBootstrapper(cmdlineParams, StopWatch.start())
+    }
+
+    fun getTesterumTests(): List<DynamicNode> {
+        testSuiteToBeExecuted = bootstrapper.runnerModuleFactory.runnerApplication.getRunnerSuiteToBeExecuted(cmdlineParams)
+
+        val junitExecutionTree = testSuiteToBeExecuted.featuresOrTests.map { defineNode(it) }
+
+
+        if (junitExecutionTree.isEmpty()) {
+            println("WARNING: No tests were found!")
```

maybe throw an exception instead?